app/models/alchemy/page/page_naming.rb

```diff
@@ -99,11 +107,16 @@ def convert_url_name(name)
       end
     end
 
-    # Urlname of parent page.
-    # Returns nil, if the parent is either a language root page or the root page itself
-    def parent_urlname
-      return if parent.nil? || parent.language_root? || parent.root?
-      parent.urlname
+    def nested_url_name(value)
+      (ancestor_slugs << convert_url_name(value)).join('/')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.